source/Token.cpp

```diff
@@ -180,6 +180,12 @@ bool Token::hasTrivia(TriviaKind triviaKind) const {
     return false;
 }
 
+bool Token::isKeywordOrIdentifier() const {
```

Comment:
Rather than checking the text, let's just compare against the set of keywords that are actually known. Just copy the switch / case blocks from getTokenKindText() and return true if it's any of the keywords and false if it's not. That means we need only one switch lookup (probably turned into a jump table by the compiler) to do the check.

See SyntaxFacts.cpp for a large collection of existing things that do this.